theme/snippets/image-caption-module.liquid

```diff
@@ -0,0 +1,37 @@
+{% assign image_module_images = shop.metafields.globals['image_module_image'][module.index] %}
+{% assign caption = shop.metafields.globals['image_module_caption'][module.index] %}
+{% assign variant = shop.metafields.globals['image_module_variant'][module.index] %}
+{% assign color = {% if styling == 'black-bg' %}'color-white'{% else %} 'color-black'{% endif %} %}
+
+{% for image in image_module_images %} 
```

Comment:
I must be missing something, but how exactly does this work? Should it not be wrapping the `if` statements below? 

The way I read it is it will iterate through each image in the image_module_images array, assign image_module_image to the last one in the array and then render the variant for that image

Reply:
The module has an image property, which is an array of images, even if you only have one image. In this field they can only upload one image, so here I'm setting the `image`  variable as that single image... maybe there is a better way to do it! 

Accentuate: `Important: the underlying metafield value will always be an iterable array. If just a single media is uploaded, the array will only have one entry. This allows for consistent use of Liquid's array filters such as first, last, sort, where etc.`

Reply:
What has been passed down to this snippet, `image-caption-module`, is the reference to that global object.

Below when rendering the variants, we don't need to iterate because we accessed that specific object by using its index here on lines 1-3:

```
{% assign image_module_images = shop.metafields.globals['image_module_image'][module.index] %}
{% assign caption = shop.metafields.globals['image_module_caption'][module.index] %}
{% assign variant = shop.metafields.globals['image_module_variant'][module.index] %}
```

Reply:
Let me know if that makes sense 🤯

Reply:
Yup, makes sense. Given that you don't really needs the `image_module_images` array for anything, maybe you can just do:

```
{% assign image_module_image = shop.metafields.globals['image_module_image'][module.index].first %}
```